docs_src/design/adr/security/0026-authentication.md

```diff
@@ -0,0 +1,296 @@
+# EdgeX Microservice Authentication
+
+## Status
+
+DRAFT
+
+## Context
+
+The AS-IS Architecture figure below depicts the current state of
+microservice communication security as of the EdgeX Levski release,
+when security is enabled:
+
+![AS-IS Architecture](0026-as-is.jpg)
+
+As shown in the diagram,
+many of the foundational services used by EdgeX Foundry
+have already been secured:
+
+* Communication with EdgeX's secret store, as implemented by
+  Hashicorp Vault, is secured over a local HTTP socket with
+  token-based authentication.  An access control list limits access
+  to the keyspace of the key value store.
+
+* Communication with EdgeX's service registry and configuration provider,
+  as implemented by Hashicorp Consul, is secured over a local HTTP
+  socket with token-based authentication,
+  with the token being mediated by Hashicorp Vault.
+  An access control list limits access to the keyspace of
+  the configuration store.
+
+* Communication with EdgeX's default database, Redis, is secured using
+  username/password authentication, with the password stored
+  in Hashicorp Vault.  An access control list limits the commands
+  that clients are allowed to issue to the server.
+
+External access to EdgeX microservices has also been secured.
+EdgeX microservices only bind to local ports,
+and are only exposed externally through a Kong API gateway.
+This gateway is configured to use TLS 1.3,
+using JWT authentication using public key cryptography.
+All external requests are filtered at the API gateway.
+URL rewriting is used to concentrate microservices
```

Comment:
Probably me, but this sentence confuses me.  Do you mean, URL rewriting is used to concentrate all outside HTTP/REST access/requests to EdgeX microservices to a single HTTP-accessible port?

Reply:
Extra words added.